src/Catalog/Registration/RegistrationEntryKey.cs

```diff
@@ -1,18 +1,23 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using NuGet.Services.Metadata.Catalog.Helpers;
+
 namespace NuGet.Services.Metadata.Catalog.Registration
 {
     public class RegistrationEntryKey
     {
+        private readonly string _normalizedVersion;
+
         public RegistrationEntryKey(RegistrationKey registrationKey, string version)
         {
             RegistrationKey = registrationKey;
             Version = version;
+            _normalizedVersion = NuGetVersionUtility.NormalizeVersion(version).ToLowerInvariant();
```

Comment:
I wonder if ToLowerInvariant should go into a more common location, so all the code converts a version  to string in the same way